openedx/core/lib/request_utils.py

```diff
@@ -120,34 +120,32 @@ def __call__(self, request):
 
     def process_exception(self, request, exception):
         """
-        Add logging and monitoring of expected errors.
+        Add logging and monitoring of ignored errors.
         """
         _log_and_monitor_expected_errors(request, exception, 'middleware')
 
 
-# .. setting_name: EXPECTED_ERRORS
+# .. setting_name: IGNORED_ERRORS
 # .. setting_default: None
-# .. setting_description: Used to configure logging and monitoring for expected errors.
+# .. setting_description: Used to configure logging and monitoring for ignored errors.
 #     This setting is configured of a list of dicts. See setting and toggle annotations for
-#     ``EXPECTED_ERRORS[N]['XXX']`` for details of each item in the dict.
-#     If this setting is a non-empty list, all uncaught errors processed will get a ``checked_error_expected_from``
-#     attribute, whether they are expected or not. Those errors that are processed and match a 'MODULE_AND_CLASS'
-#     (documented elsewhere), will get an ``error_expected`` custom attribute. Unexpected errors would be errors with
-#     ``error_expected IS NULL``. For additional diagnostic information for ignored errors, see the
-#     EXPECTED_ERRORS[N]['IS_IGNORED'] annotation.
+#     ``IGNORED_ERRORS[N]['XXX']`` for details of each item in the dict.
+#     If this setting is a non-empty list, all uncaught errors processed will get a ``checked_error_ignored_from``
+#     attribute, whether they are ignored or not. Those errors that are processed and match a 'MODULE_AND_CLASS'
+#     IGNORED_ERRORS[N]['IS_IGNORED'] annotation.
 # .. setting_warning: We use Django Middleware and a DRF custom error handler to find uncaught errors. Some errors may
-#     slip through the cracks, like ValidationError. Any error where ``checked_error_expected_from IS NULL`` is
+#     slip through the cracks, like ValidationError. Any error where ``checked_error_ignored_from IS NULL`` is
 #     an error that was not processed.
 
-# .. setting_name: EXPECTED_ERRORS[N]['MODULE_AND_CLASS']
+# .. setting_name: IGNORED_ERRORS[N]['MODULE_AND_CLASS']
 # .. setting_default: None
-# .. setting_description: Required error module and class name that is expected. For example,
+# .. setting_description: Required error module and class name that is ignored. For example,
 #     ``rest_framework.exceptions.PermissionDenied``.
```

Comment:
Because `IGNORED_ERRORS[N]['IS_IGNORED']` is going away, I'm moving the details that were captured there to this setting. Please update as follows:
```suggestion
#     ``rest_framework.exceptions.PermissionDenied``. If the current error matches the module and class
#      defined here, the middleware will add the custom attributes ``error_ignored_class`` and ``error_ignored_message``
#.     to help diagnose issues with ignored errors, since this data is not otherwise available.
#      For example of ignoring errors in New Relic, see:
#      https://docs.newrelic.com/docs/agents/manage-apm-agents/agent-data/manage-errors-apm-collect-ignore-or-mark-expected/#ignore  pylint: disable=line-too-long,useless-suppression
#      To query for ignored errors, you would use ``error_ignored_class IS NOT NULL``.
# .. setting_warning: At this time, an error that matches won't actually be ignored. These settings should be set to match
#     the ignored error configuration found elsewhere, like in New Relic. When monitoring, no errors should ever have the attribute
#     ``error_ignored_class``. Only Transactions should have this custom attribute. If found for an error, it means we
#     are stating an error should be ignored when it is not actually configured as such, or the (New Relic) configuration is not
#     working.
```

Reply:
This change was not completed. This is meant to be an update for this comment.